Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,101 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+
+        private string _filePath;
+
+        public SampleData()
+        {
+            _filePath = "People.csv";
+        }
+
+        public SampleData(string filePath)
+        {
+            _filePath = filePath!;
+        }
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows
+        {
+            get
+            {
+                return File.ReadAllLines(_filePath).Skip(1);
+            }
+        }
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            IEnumerable<string> stateList = 
+                (
+                    from line in CsvRows
+                    let state = line.Split(',')
+                    orderby state[6]
+                    select state[6]
+                )
+                .Distinct().ToList();
+
+            return stateList;
+        }
 
         // 3.
         public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+        {
+            string[] states = GetUniqueSortedListOfStatesGivenCsvRows().Select(item => item).ToArray();

```

Comment:
```suggestion
            string[] states = GetUniqueSortedListOfStatesGivenCsvRows().ToArray();
```
You don't need this select here.